sentinel-extension/sentinel-annotation-aspectj/src/main/java/com/alibaba/csp/sentinel/annotation/aspectj/SentinelResourceAspect.java

```diff
@@ -59,12 +59,33 @@ public Object invokeResourceWithSentinel(ProceedingJoinPoint pjp) throws Throwab
         } catch (BlockException ex) {
             return handleBlockException(pjp, annotation, ex);
         } catch (Throwable ex) {
-            Tracer.trace(ex);
+            if (isTrackedException(ex, annotation.exceptionsToTrace())) {
+                Tracer.trace(ex);
+            }
             throw ex;
         } finally {
             if (entry != null) {
                 entry.exit();
             }
         }
     }
+
+    /**
+     * whether the exception is in tracked list of exception classes
+     *
+     * @param ex
+     * @param exceptionsToTrace
+     * @return
+     */
+    private boolean isTrackedException(Throwable ex, Class<? extends Throwable>[] exceptionsToTrace) {
+        if (exceptionsToTrace == null) {
+            return false;
+        }
+        for (Class exceptionToTrace : exceptionsToTrace) {
+            if (ex.getClass().isAssignableFrom(exceptionToTrace)) {
```

Comment:
It's incorrect. For each exception to trace, it should be:

```suggestion
            if (exceptionToTrace.isAssignableFrom(ex.getClass())) {
```

**Please be sure to add test cases and run the demo to verify it.**